lib/Twig/Environment.php

```diff
@@ -313,6 +315,10 @@ public function getTemplateClass($name, $index = null)
         $key .= json_encode(array_keys($this->extensions));
         $key .= function_exists('twig_template_get_attributes');
         $key .= ':'.PHP_MAJOR_VERSION.':'.PHP_MINOR_VERSION;
+        $key .= ':debug:'.($this->debug ? '1' : '0');
+        $key .= ':charset:'.$this->charset;
```

Comment:
AFAIK, this is useless: all Twig filters needing the charset are getting it from the environment at runtime. They cannot get it injected at compile-time